src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -98,36 +106,95 @@ p5.RendererGL.prototype.vertex = function(){
  */
 p5.RendererGL.prototype.endShape =
 function(mode, isCurve, isBezier,isQuadratic, isContour, shapeKind){
-
-  var gl = this.GL;
-  var shader = this.curShader;
-  if (shader === this._getColorShader()) {
-    // this is the fill/stroke shader for retain mode.
+  if(this.drawMode !== constants.TEXTURE) {
     // must switch to immediate mode shader before drawing!
-    shader = this.setShader(this._getImmediateModeShader());
-
+    this.setFillShader(this._getImmediateModeShader());
     // note that if we're using the texture shader...
     // this shouldn't change. :)
   }
-  shader.bindShader();
+
+  if(this.curStrokeShader.active === true) {
+    for(var i=0; i<this.immediateMode.vertexPositions.length; i++) {
+      if(i+1 < this.immediateMode.vertexPositions.length) {
+        this.immediateMode.edges.push([i, i+1]);
+      } else {
+        this.immediateMode.edges.push([i, 0]);
+      }
+    }
+    this._edgesToVerticesImmediateMode();
+    this._drawStrokeImmediateMode();
+  }
+  if(this.curFillShader.active === true) {
+    this._drawFillImmediateMode(mode, isCurve, isBezier,isQuadratic,
+      isContour, shapeKind);
+  }
+  //clear out our vertexPositions & colors arrays
+  //after rendering
+  this.immediateMode.vertexPositions.length = 0;
+  this.immediateMode.vertexColors.length = 0;
+  this.immediateMode.uvCoords.length = 0;
+  this.isImmediateDrawing = false;
+
+  return this;
+};
+
+/**
+ * Create 4 vertices for each stroke line, two at the beginning position
+ * and two at the end position. These vertices are displaced relative to
+ * that line's normal on the GPU
+ * @return {p5.Geometry}
+ */
+p5.RendererGL.prototype._edgesToVerticesImmediateMode = function() {
```

Comment:
this code looks very similar to _edgesToVertices() which makes me wonder if it can be abstracted somehow rather than duplicating the code?